### DISCUSSION TITLE
how to get script real line number?

### DISCUSSION THREAD

**Discussion by yundiz** [dop251/goja#598] [General]
hi,
 i want to add debug function into goja ,and i can't find real line number in compile.* and vm.* file,
in the following code, i get the source code line number if current program srcPos=codeLine, add wait the next  command to debug code, is that possible?

![Snipaste_2024-08-16_13-40-12](https://github.com/user-attachments/assets/52be27df-921d-4b89-9cb4-308794e3808e)

**Comment by yundiz** [comment]
the following picture show the line and srcPos，but  my script code just have 3 lines

![Snipaste_2024-08-16_13-46-40](https://github.com/user-attachments/assets/a2c39036-f8f7-403b-b1c7-35ec0dd61fc6)

**Comment by dop251** [comment]
You should probably have a look at [StackFrame.Position() method](https://github.com/dop251/goja/blob/8130cadc5774d6c30ecbef6e9cb07c406afcdcc5/runtime.go#L223)
